GDJS/Runtime/howler-sound-manager/howler-sound-manager.js

```diff
@@ -14,87 +14,146 @@
+    // sync'ed with the sound - though this should be redundant
+    // with `play`/`pause` methods already doing that. Keeping
+    // that to be sure that the status is always correct.
+    this._howl.on('play', function () {
+      that._paused = false;
+      that._stopped = false;
+    });
+    this._howl.on('pause', function () {
+      that._paused = true;
+      that._stopped = false;
+    });
+
+    // In case the howl didn't preload, start the sound once it finishes loading.
+    this._howl.on('load', function () {
+      that.play();
+    });
+  };
 
-// Redefine `stop`/`play`/`pause` to ensure the status of the sound
-// is immediately updated (so that calling `stopped` just after
-// `play` will return false).
+  play() {
+    this._paused = false;
+    this._stopped = false;
+    this._id = this._howl.play(this._id === null ? undefined : this._id);
+    return this._id;
+  }
 
-gdjs.HowlerSound.prototype.stop = function () {
-  this._paused = false;
-  this._stopped = true;
-  return Howl.prototype.stop.call(this);
-};
-gdjs.HowlerSound.prototype.play = function () {
-  this._paused = false;
-  this._stopped = false;
-  return Howl.prototype.play.call(this);
-};
-gdjs.HowlerSound.prototype.pause = function () {
-  this._paused = true;
-  this._stopped = false;
-  return Howl.prototype.pause.call(this);
-};
+  pause() {
+    if(this._id === null) return;
+    this._paused = true;
+    this._stopped = false;
+    return this._howl.pause(this._id);
+  }
 
-// Add methods to query the status of the sound:
+  stop() {
+    if(this._id === null) return;
+    this._paused = false;
+    this._stopped = true;
+    return this._howl.stop(this._id);
+  }
 
-gdjs.HowlerSound.prototype.paused = function () {
-  return this._paused;
-};
-gdjs.HowlerSound.prototype.stopped = function () {
-  return this._stopped;
-};
-gdjs.HowlerSound.prototype.canBeDestroyed = function () {
-  return this._canBeDestroyed;
-};
+  playing() {
+    if(this._id === null) return false;
+    return this._howl.playing(this._id);
+  }
+
+  paused() {
+    return this._paused
+  }
 
-// Methods to safely update the rate of the sound:
+  stopped() {
+    return this._stopped
+  }
 
-gdjs.HowlerSound.prototype.getRate = function () {
-  return this._rate;
-};
-gdjs.HowlerSound.prototype.setRate = function (rate) {
-  this._rate = gdjs.HowlerSoundManager.clampRate(rate);
-  this.rate(this._rate);
+  canBeDestroyed() {
+    return this._canBeDestroyed
+  }
+
+  getRate() {
+    return this._rate;
+  }
+
+  setRate(rate) {
+    if(this._id === null) return;
+    this._rate = gdjs.HowlerSoundManager.clampRate(rate);
+    return this._howl.rate(this._rate, this._id);
+  }
+
+  rate(rate) {
+    return this.setRate(rate);
+  }
+
+  loop(loop) {
+    if(this._id === null) return;
+    return this._howl.loop(loop, this._id);
+  }
+
+  volume(volume) {
+    if(this._id === null) return;
+    return this._howl.volume(volume, this._id);
+  }
+
+  mute(mute) {
```

Comment:
I thought those might be useful for a javascript user. If they see howler sound, and it has almost the same api as a Howl, they might be confused why this doesn't work. I also think it might be nice to add those features to GDevelop.
And `on` is actually used.

Reply:
Fair enough then!

Reply:
Maybe we should write a few jsdocs for these functions then if we want them to show in the generated documentation?